lib/Runner.js

```diff
@@ -52,98 +57,74 @@ class Runner {
       });
   }
 
-  static _toMilliseconds(time) {
-    const nanoseconds = (time[0] * 1000000000) + time[1];
-    return Math.round(nanoseconds / 1000000);
-  }
-
   static _getAverage(durations) {
     const average = durations.reduce((acc, duration) => acc + duration, 0) / durations.length;
     return Math.round(average);
   }
 
   static _parseCycleResult(cycleResult) {
-    const meaningfullResults = cycleResult.slice(1); // first result is usually influenced by cold start
+    const percentiles = PERCENTILES.reduce((memo, i) => {
+      return Object.assign(memo, { [i]: percentile(i, cycleResult) });
+    }, {});
 
     return {
-      min: Math.min(...meaningfullResults),
-      max: Math.max(...meaningfullResults),
-      avg: Runner._getAverage(meaningfullResults)
+      min: Math.min(...cycleResult),
+      max: Math.max(...cycleResult),
+      avg: Runner._getAverage(cycleResult),
+      percentiles
     };
   }
 
-  _cycle(count, acc) {
-    if (count <= 0) {
-      return BPromise.resolve(acc);
-    }
-
-    if (!acc) {
-      acc = []; // eslint-disable-line no-param-reassign
-    }
-
-    const self = this;
-    return BPromise
-      .delay(INVOCATION_DELAY)
+  _cycle(count) {
+    return BPromise.resolve()
       .then(() => {
-        return typeof self.payload === 'function'
-          ? self.payload(acc.length + 1)
-          : self.payload;
+        if (this.warmup) {
```

Comment:
Warmup looks useful only with concurrency set to 1, otherwise Lambda would be randomly starting new containers.